AppCenter/AppCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -264,100 +277,153 @@ + (void)enableAutoVacuumInOpenedDatabase:(void *)db {
+    [MSDBStorage executeSelectionQuery:@"PRAGMA auto_vacuum = FULL;" inOpenedDatabase:db withValues:nil];
+    [MSDBStorage executeSelectionQuery:@"VACUUM;" inOpenedDatabase:db withValues:nil];
   }
 }
 
-- (NSUInteger)countEntriesForTable:(NSString *)tableName condition:(nullable NSString *)condition {
+- (NSUInteger)countEntriesForTable:(NSString *)tableName
+                         condition:(nullable NSString *)condition
+                        withValues:(nullable MSStorageBindableArray *)values {
   NSMutableString *countLogQuery = [NSMutableString stringWithFormat:@"SELECT COUNT(*) FROM \"%@\" ", tableName];
   if (condition.length > 0) {
     [countLogQuery appendFormat:@"WHERE %@", condition];
   }
-  NSArray<NSArray<NSNumber *> *> *result = [self executeSelectionQuery:countLogQuery];
+  NSArray<NSArray<NSNumber *> *> *result = [self executeSelectionQuery:countLogQuery withValues:values];
   return (result.count > 0) ? result[0][0].unsignedIntegerValue : 0;
 }
 
++ (int)executeNonSelectionQuery:(NSString *)query inOpenedDatabase:(void *)db {
+  return [self executeNonSelectionQuery:query inOpenedDatabase:db withValues:nil];
+}
+
 - (int)executeNonSelectionQuery:(NSString *)query {
+  return [self executeNonSelectionQuery:query withValues:nil];
+}
+
+- (int)executeNonSelectionQuery:(NSString *)query withValues:(nullable MSStorageBindableArray *)values {
   return [self executeQueryUsingBlock:^int(void *db) {
-    return [MSDBStorage executeNonSelectionQuery:query inOpenedDatabase:db];
+    return [MSDBStorage executeNonSelectionQuery:query inOpenedDatabase:db withValues:values];
   }];
 }
 
-+ (int)executeNonSelectionQuery:(NSString *)query inOpenedDatabase:(void *)db {
-  char *errMsg = NULL;
-  int result = sqlite3_exec(db, [query UTF8String], NULL, NULL, &errMsg);
-  if (result == SQLITE_CORRUPT || result == SQLITE_NOTADB) {
-    MSLogError([MSAppCenter logTag], @"A database file is corrupted: %d - %@", result, [NSString stringWithUTF8String:errMsg]);
-  } else if (result == SQLITE_FULL) {
-    MSLogDebug([MSAppCenter logTag], @"Query failed with error: %d - %@", result, [NSString stringWithUTF8String:errMsg]);
-  } else if (result != SQLITE_OK) {
-    MSLogError([MSAppCenter logTag], @"Query \"%@\" failed with error: %d - %@", query, result, [NSString stringWithUTF8String:errMsg]);
++ (int)executeNonSelectionQuery:(NSString *)query inOpenedDatabase:(void *)db withValues:(nullable MSStorageBindableArray *)values {
+  return [MSDBStorage executeQuery:query
+                  inOpenedDatabase:db
+                        withValues:values
+                        usingBlock:^(void *statement) {
+                          int stepResult = sqlite3_step(statement);
+                          if (stepResult == SQLITE_DONE) {
+                            return SQLITE_OK;
+                          }
+                          NSString *errorMessage = [NSString stringWithUTF8String:sqlite3_errmsg(db)];
```

Comment:
Do you think that we can use `sqlite3_errstr` now that we support iOS9+? Context: https://github.com/microsoft/appcenter-sdk-apple/pull/656

Reply:
No need for `sqlite3_errstr` here, `sqlite3_errmsg` is the right one

Reply:
Why @MatkovIvan? From what I could read `sqlite3_errstr(stepResult)` would do the utf8 conversion for us.

Reply:
But nitpick anyway.